packages/circuit-ui/hooks/useClickOutside/useClickOutside.docs.mdx

```diff
@@ -0,0 +1,21 @@
+import { Status, Story } from '../../../../.storybook/components';
+
+# useClickOutside()
+
+<Status.Stable />
+
+Calls a function when clicking outside an element.
+
+<Story id="hooks-useclickoutside--example" />
+
+```ts
+useClickOutside(
+  ref: RefObject<HTMLElement>,
+  callback: (event: KeyboardEvent) => void,
+  active = true,
+): void;
+```
+
+## Usage
+
+Application menus, modals, and other components with an "open" state should be dismissable by clicking outside. This hook enables this behaviour. Its event listener can be deactivated when the component isn't active (i.e. closed) to improve performance.
```

Comment:
Should we mention that it shouldn't be manually implemented for existing components like the Popover or the Modal? (and that these components should be used when possible over implementing custom components from scratch)?

IMO exporting these hooks is fine but it shouldn't be seen as a message that building your own dropdown or modal is encouraged, there is much more to making this kind of components accessible (think markup etc.)

Reply:
That's a very good point. I'll add a note about it 👍🏻